change letter case for test case identifier (#1200)
zonemaster/zonemaster: docs/internal/templates/specifications/tests/TestCaseIdentifierSpecification.md

# Test Case Identifier Specification

> All test cases belong to one module, and it gets its name from that module's
> name. The following modules are defined and available:

Reviewer: ```suggestion
All test cases belong to one module and their names are based on that module’s name.
The following modules are defined and available:
```

Author: I changed the wording to match your proposal

> name. The following modules are defined and available:

Reviewer 2: This document now refer "modules" but previously it referred to "test levels". We should use the same terminology in all places. The reason why this document referred to "test level" is because it is called so in other documents. We should be consistent.

See e.g. https://github.com/zonemaster/zonemaster/blob/master/docs/public/specifications/tests/MasterTestPlan.md#definition-of-test-levels

Author: I think the current wording across component is Module,

* Test zone data : https://github.com/zonemaster/zonemaster/blob/397c1281da08d0f3323fae6eae1ab54ec0779838/test-zone-data/README.md?plain=1#L94C1-L96C62
* Backend RPCAPI document : https://github.com/zonemaster/zonemaster/blob/397c1281da08d0f3323fae6eae1ab54ec0779838/docs/public/using/backend/rpcapi-reference.md?plain=1#L328
* CLI usage : https://github.com/zonemaster/zonemaster-cli/blob/master/lib/Zonemaster/CLI.pm#L119-L127

and so on.

* Syntax
* Zone

The module name is not case sensitive, but camel case, as defined above,

> should be prefered.

Reviewer 2: ```
The module name is not case sensitive, but the forms defined above,
should be used when referring to the modules, i.e. "Address" and neither
"ADDRESS" nor "address".
```

Author: done.

The test case identifier in the test case specification (both in the main
headline and in the "Test case identifier" section) uses the module name,
as defined above, and has the format: `{Module name} + {Index}`

When referencing to a test case the module name may be used in other letter

> case, but for readability, camel case sould be prefered.

Reviewer 2: I do not think these are examples of "camelCase", https://en.wikipedia.org/wiki/Camel_case

"sould" --> "should"

Reviewer 2: No, "Address" is not PascalCase.

Author: See my other comment.

The `{Index}` is a two-digit suffix 01-99, and should be selected so that the test
case identifier is unique. Normally the first free index is selected.

Example of test case identifiers:

* Address03
* Basic04
* DNSSEC15
* Zone06

> One exceptional test case identifier is `Basic00`, in which the index

Reviewer 2: Basic00 has been removed, hasn't it?

Author: Not yet
